crates/net/src/sim2h_worker.rs

```diff
@@ -333,6 +335,42 @@ impl Sim2hWorker {
         }
     }
 
+    fn self_store_authored_aspects(&mut self) {
+        if !self.has_self_stored_authored_aspects
+            && self.initial_gossiping_list.is_some()
+            && self.initial_authoring_list.is_some()
+        {
+            let authoring_list = self.initial_authoring_list.take().unwrap();
```

Comment:
minor point, but this is weird and the kind of thing clippy doesn't like, what would it mean for a list to be `None` vs. `Some`-yet-empty, and then why do we do an `is_some()` in the condition but then need to `unwrap()` in the body? @lucksus

Reply:
Yeah this could have been written where the lists are `take`n and bound to local vars before the condition, to avoid the unwrap. But, minor.

The some-ness indicates whether or not the list got returned from the network, but if it's empty, that's just what was fetched.

Reply:
@maackle ah, i see, yeah that does make sense "unknown" vs. "known and empty"